src/mlpack/methods/ann/init_rules/network_init.hpp

```diff
@@ -20,6 +20,57 @@
 
 namespace mlpack {
 
+/**
+ * Create a temporary matrix with a specific offset, this is only necessary
+ * since we have two  backend libraries to support. The first function
+ * overload is called when the MatType == arma.
+ *
+ * @param tmp The returned temporary matrix.
+ * @param Mat The original matrix we are taking part from it.
+ * @param offset The Start point of the tmp matrix.
+ * @param numRows The number of rows of the tmp matrix.
+ * @param numCols The numbers or cols of the tmp matrix.
+ */
+template<typename MatType,
+         typename = typename std::enable_if<
+         !IsCootMatType<MatType>::value>::type>
+void MakeTmp(MatType& tmp,
+             const MatType& Mat,
+             const size_t offset,
+             const size_t numRows,
+             const size_t numCols)
+{
+  tmp = MatType(Mat.memptr() + offset, numRows, numCols, false,
+      false);
+}
+
+#ifdef MLPACK_HAS_COOT
```

Comment:
This guard isn't necessary unless you specifically try to reference a `coot::` class.  Here you only use `.get_dev_mem()` so it is fine.